packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -63,9 +65,22 @@ func GetProtocolStateVariables(
 	opts.Context, cancel = CtxWithTimeoutOrDefault(opts.Context, defaultTimeout)
 	defer cancel()
 
+	var slotBV1 bindings.TaikoDataSlotBV1
 	slotA, slotB, err := taikoL1Client.GetStateVariables(opts)
 	if err != nil {
-		return nil, err
+		if errors.Is(err, ErrSlotBMarshal) {
+			slotA, slotBV1, err = taikoL1Client.GetStateVariablesV1(opts)
```

Comment:
Shall we also check the error here?

Reply:
I think there is no need.
If we fail both ways, just throw an error!

Reply:
If we don't check the `err` here, `slotBV1` / `slotA` will be `nil` and the code below will panic?

Reply:
Yes, fixed